ci/it/testcases/test_ab.go

```diff
@@ -0,0 +1,287 @@
+
+	// this is very deterministic log
+
+	// Fixed start time
+	start, err := time.Parse(time.RFC3339, "2019-01-22T03:56:14+03:30")
+	if err != nil {
+		t.Fatalf("Failed to parse time: %s", err)
+	}
+
+	for i := 0; i < 100; i++ {
+
+		m := make(map[string]interface{})
+
+		m["remote_addr"] = "127.0.0.1"
+		m["request"] = fmt.Sprintf("GET /test-%d", i)
+		m["status"] = ((i % 4) + 2) * 100 // 200,300,400,500
+		m["body_bytes_sent"] = i * 1000
+		m["http_referer"] = "-"
+
+		if i%2 == 0 {
+			m["http_user_agent"] = "Mozilla/5.0"
+		} else {
+			m["http_user_agent"] = "curl/7.68"
+		}
+		m["@timestamp"] = start.Add(time.Duration(i) * time.Second).Format(time.RFC3339)
+
+		doc, err := json.Marshal(m)
+		if err != nil {
+			t.Fatalf("Failed to marshal json: %s", err)
+		}
+
+		resp, body := a.RequestToQuesma(ctx, t, "POST", "/test_index/_doc", doc)
+
+		if resp.StatusCode != http.StatusOK {
+			fmt.Println("Failed POST request: ", string(body))
+			t.Fatalf("Failed to make POST request: %s", resp.Status)
+		}
+		_ = resp.Body.Close()
+
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+	}
+
+	chQuery := "SELECT count(*) FROM test_index"
+	rows, err := a.ExecuteClickHouseQuery(ctx, chQuery)
+	if err != nil {
+		t.Fatalf("Failed to execute query: %s", err)
+	}
+
+	var count int
+	for rows.Next() {
+		if err := rows.Scan(&count); err != nil {
+			t.Fatalf("Failed to scan row: %s", err)
+		}
+	}
+
+	if rows.Err() != nil {
+		t.Fatalf("Failed to scan row: %s", rows.Err())
+	}
+
+	assert.Equal(t, 100, count)
+
+	_, _ = a.RequestToElasticsearch(ctx, "GET", "/test_index/_refresh", nil)
+
+	// Also make sure no such index got created in Elasticsearch
+	resp, err := a.RequestToElasticsearch(ctx, "GET", "/test_index/_count", nil)
+	if err != nil {
+		t.Fatalf("Failed to make GET request: %s", err)
+	}
+	defer resp.Body.Close()
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("Failed to read response body: %s", err)
+	}
+
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	elasticResult := make(map[string]interface{})
+	if err := json.Unmarshal(bodyBytes, &elasticResult); err != nil {
+		t.Fatalf("Failed to unmarshal response: %s", err)
+	}
+
+	assert.Equal(t, 100, int(elasticResult["count"].(float64)))
+
+}
+
+func (a ABTestcase) waitForAbResults(ctx context.Context, t *testing.T) {
+
+	timeout := 10
+
+	for i := 0; i <= timeout; i++ {
+		time.Sleep(1 * time.Second)
+
+		if i == timeout {
+			t.Fatalf("Didn't find any results in ab_testing_logs table. Failing the test.")
+		}
+
+		count := "SELECT count(*) FROM ab_testing_logs"
+		rows, err := a.ExecuteClickHouseQuery(ctx, count)
+
+		if err != nil {
+			if strings.Contains(err.Error(), "code: 60") { // 60 is the code for table not found
+				continue
+			}
+
+			t.Fatalf("Failed to execute query: %s", err)
+		}
+		for rows.Next() {
+			var c int
+			if err := rows.Scan(&c); err != nil {
+				t.Fatalf("Failed to scan row: %s", err)
+			}
+			if c > 0 { // we have some results,
```

Comment:
Shouldn't this wait for an exact number of results (number of JSON files - skipped count)?

Reply:
OK, I see now that `ab_testing_logs` is cleared after each JSON file, so this logic makes sense.